apps/vr-tests-react-components/src/stories/Positioning.stories.tsx

```diff
@@ -0,0 +1,487 @@
+    },
+  },
+
+  gridWrapper: {
+    width: '400px',
+    height: '400px',
+    display: 'grid',
+    gridTemplateColumns: 'auto auto auto',
+    columnGap: '20px',
+    rowGap: '50px',
+  },
+
+  boundary: {
+    ...shorthands.border('2px', 'dashed', 'red'),
+  },
+
+  box: {
+    ...shorthands.padding('15px'),
+    ...shorthands.border('1px', 'solid', 'blue'),
+    backgroundColor: 'white',
+  },
+
+  arrow: {
+    ...createArrowStyles({ arrowHeight: 5 }),
+    backgroundColor: 'red',
+  },
+
+  seeThrough: {
+    opacity: 0.6,
+  },
+});
+
+const positions = [
+  ['above', 'start'],
+  ['above', 'center'],
+  ['above', 'end'],
+  ['below', 'start'],
+  ['below', 'center'],
+  ['below', 'end'],
+  ['before', 'top'],
+  ['before', 'center'],
+  ['before', 'bottom'],
+  ['after', 'top'],
+  ['after', 'center'],
+  ['after', 'bottom'],
+] as const;
+
+const Box = React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivElement>>((props, ref) => {
+  const styles = useStyles();
+  return (
+    <div {...props} className={mergeClasses(styles.box, props.className)} ref={ref}>
+      {props.children}
+    </div>
+  );
+});
+
+const PositionAndAlignProps: React.FC<{ positionFixed?: boolean }> = ({ positionFixed }) => {
+  const styles = useStyles();
+  const positionedRefs = positions.reduce<ReturnType<typeof usePopper>[]>((acc, cur) => {
+    const popperOptions: PositioningProps = { position: cur[0], align: cur[1] };
+    // positionFixed is not public yet
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    popperOptions.positionFixed = positionFixed;
+
+    // this loop is deterministic
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const popperRefs = usePopper(popperOptions);
+    acc.push(popperRefs);
+    return acc;
+  }, []);
+
+  const targetRef = useMergedRefs(...positionedRefs.map(x => x.targetRef));
+
+  return (
+    <div className={styles.wrapper}>
+      {positions.map(([position, align], i) => (
+        <Box key={`${position}-${align}`} ref={positionedRefs[i].containerRef}>{`${position}-${align}`}</Box>
+      ))}
+      <div ref={targetRef} className="target" />
+    </div>
+  );
+};
+
+const Offset = () => {
+  const styles = useStyles();
+  const positionedRefs = positions.reduce<ReturnType<typeof usePopper>[]>((acc, cur) => {
+    // this loop is deterministic
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const popperRefs = usePopper({ position: cur[0], align: cur[1], offset: [10, 10] });
+    acc.push(popperRefs);
+    return acc;
+  }, []);
+
+  const targetRef = useMergedRefs(...positionedRefs.map(x => x.targetRef));
+
+  return (
+    <div className={styles.wrapper}>
+      {positions.map(([position, align], i) => (
+        <Box key={`${position}-${align}`} ref={positionedRefs[i].containerRef}>{`${position}-${align}`}</Box>
+      ))}
+      <div ref={targetRef} className="target" />
+    </div>
+  );
+};
+
+const OffsetFunction = () => {
```

Comment:
Could you please explain the `y` difference between RTL and LTR screenshots for this story?
<img width="554" alt="image" src="https://user-images.githubusercontent.com/9615899/155293458-ebe5b5eb-0c82-41cd-845f-bfbd70bf1138.png">

Reply:
The offset modifier in popper v2 is not configurable enough to take into account the placement and only accepts `offset` as a function or scalar value. Therefore we simply invert the cross axis offset for every placement, where we **should** actually be modifying the cross axis offset **only for vertical placements** .The same bug is in Northstar too : https://github.com/microsoft/fluentui/blob/f8cf163d50469e9457aab009ec6f1801a9da804b/packages/react-positioning/src/usePopper.ts#L81

We can fix this with Floating UI upgrade (#21832) since the middleware pattern allows us to finally use `placement` as an extra configuration for RTL offsets.

Reply:
Tracked in #21840